### PR TITLE
move discourse prod s3 permissions from custom role to nodegroup roles

### PR DESCRIPTION
**Pull request by cmharlow**
Jira: Part of https://mozilla-hub.atlassian.net/browse/SE-2172

What this PR does:
* Moves the discourse prod s3 upload permissions from using a specific iam role to applying the permissions to the nodegroup role
* using a specific iam role is no longer feasible in the new clusters, since we don't use aws-iam add-ons to kubernetes

Note: this was applied already to handle the DNS migration / switch over of production discourse, as the uploads error was caught in post-migration testing.

### CODE REVIEW

File: terraform/applications/discourse/iam.tf [hunk 1]
```diff
@@ -1,39 +1,14 @@
-resource "aws_iam_role" "discourse_role" {
-  name               = "discourse-${var.environment}"
-  path               = "/discourse/"
-  assume_role_policy = data.aws_iam_policy_document.allow_assume_role.json
+locals {
+  cluster_workers_role_name = replace(
+    data.terraform_remote_state.k8s.outputs.cluster_worker_iam_role_arn,
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/",
+    ""
+  )
```
Comment (smarnach) [inline terraform/applications/discourse/iam.tf:6]: Our EKS module [already exports the role name](https://github.com/mozilla-it/terraform-modules/blob/33c9f55b23035e130291e13f0f4bab542736727b/aws/eks/outputs.tf#L37-L39), so if we re-export the name [in the shared Terraform code in this repo](https://github.com/mozilla-it/itse-apps-prod-1-infra/blob/083c744e7760f1a370ef776adfeeac78700af50a/terraform/shared/outputs.tf) we can make do without this hack.

**Review by smarnach — CHANGES_REQUESTED** [review]

File: terraform/applications/discourse/iam.tf (hunk 1 above)
Comment (cmharlow) [reply terraform/applications/discourse/iam.tf:6]: @smarnach feel free to edit the commit here and the one in it’s-apps-stage-infra-1 to show your requested changes and ensure the role names are indeed the same (we often need a workers role specifically).

Comment (smarnach) [reply terraform/applications/discourse/iam.tf]: You are of course right that the role name I linked was the cluster IAM role, not the worker IAM role. I had to file https://github.com/mozilla-it/terraform-modules/pull/160 to add the worker IAM role as an additional output.

Comment (cmharlow) [reply terraform/applications/discourse/iam.tf]: thanks @smarnach ! will follow up on this PR now, thanks for getting it all aligned for me

**Comment by cmharlow** [comment]
changes pushed by @smarnach look good, after related work in https://github.com/mozilla-it/terraform-modules/pull/160 & similar work in https://github.com/mozilla-it/itse-apps-stage-1-infra/pull/103

also had to update the psql version to 10.17 to match RDS auto-increment for minor versions changes.

**Comment by cmharlow** [comment]
tf plans & applies clean, just handing back to @smarnach and maybe @rvaralda to 👀 one more time before merge.

**Review by smarnach — APPROVED** [review]

**Pull request merged**